### PR TITLE
Fixing Ipython Notebook widgets for Py3

### PR DESCRIPTION
**Pull request by MerlinSmiles**
This goes for issue  #232, how do I label that so they get connected?

This Pull requests fixes the broken widgets for the newest IPython Notebook 
Both py3 and py2 seem to work
There are still issues with overlapping widgets though, i have no idea on how to arrange them properly, seems they only have a width attribute...

### CODE REVIEW

File: lmfit/parameter.py [hunk 1]
```diff
@@ -102,6 +102,7 @@ def _update_param(name):
             if _updated[name]:
                 return
             par = self.__getitem__(name)
+            if not hasattr(par, '_expr_ast'):   par._expr_ast = None
```
Comment (tacaswell) [inline lmfit/parameter.py]: Please do not compress if statements onto one line like this.

**Comment by danielballan** [comment]
@MerlinSmiles Thanks for taking the time to fix this. Referring to the issue as you did is enough to create a reference between them. If you want #232 to be automatically closed when this PR is merged, use the phrase: closes #232.

File: lmfit/ui/ipy_fitter.py [hunk 2]
```diff
@@ -10,21 +10,12 @@
 
 import IPython
 from IPython.display import display, clear_output
-# Widgets were only experimental in IPython 2.x, but this does work there.
```
Comment (danielballan) [inline lmfit/ui/ipy_fitter.py]: Why not keep 2.x compatibility?

File: lmfit/parameter.py (hunk 1 above)
Comment (MerlinSmiles) [inline lmfit/parameter.py]: Removed this again as it had nothing to do with the issue mentioned

File: lmfit/ui/ipy_fitter.py [hunk 3]
```diff
@@ -116,12 +107,11 @@ def close(self):
         self.max_checkbox.close()
 
     def _repr_html_(self):
-        box = Box()
+        box = HBox()
         box.children = [self.value_text, self.vary_checkbox,
                         self.min_text, self.min_checkbox,
                         self.max_text, self.max_checkbox]
         display(box)
```
Comment (danielballan) [inline lmfit/ui/ipy_fitter.py]: By putting `box.add_class('hbox')` inside an `if` block like this we could keep compatibility with IPython 2.x.

```
if IPython.release.version_info[0] == 2:
    box.add_class('hbox')
```

**Comment by MerlinSmiles** [comment]
@danielballan Ok, thanks, so I assume if i end this with the phrase it will work when it is merged at some point...

closes #323

File: lmfit/ui/ipy_fitter.py (hunk 3 above)
Comment (MerlinSmiles) [inline lmfit/ui/ipy_fitter.py]: When I test this in an py2 environment it works fine with the widgets as they are. This is what I thought would stop py2 compatibility, this is a quick test:
```
%matplotlib inline
import numpy as np
import lmfit
from lmfit.ui import Fitter
data = np.array([1,2,3,4,5,4,3,2])
x = np.arange(len(data))/2.0

Fitter( x=x, data = data)
```

Comment (danielballan) [inline lmfit/ui/ipy_fitter.py]: We're talking about IPython 2/3, not Python 2/3, right?

Comment (MerlinSmiles) [inline lmfit/ui/ipy_fitter.py]: Oops, sorry totally missed that.
I'll revert those changes, but I cant test the widgets then though...

Comment (danielballan) [inline lmfit/ui/ipy_fitter.py]: I'm not following. IPython 2.x supports widgets. Ideally, NotebookFitter should work on both Python 2.7 and 3.x and IPython 2.x and 3.x. The widget API changed between IPython 2.x and 3.x. Later releases in the 2.x series support both 2-style and 3-style widgets, but early releases (e.g., 2.0) only support 2-style. That is why I imported things in two different ways.

Comment (MerlinSmiles) [inline lmfit/ui/ipy_fitter.py]: I must admit I didnt read carefully about the python Ipython issue.
I had the same problem as in the issue mentioned above, and fixed them for me. Here I'm using Python 3.4 and IPython 3.
Then I only checked with a Python 2.7 virtual env. if that would work, but still using Ipython 3 so I missed the IPython 2.x issue, as I thought it was without the I...
I'll move the Ipython 2 stuff back in.

Comment (danielballan) [inline lmfit/ui/ipy_fitter.py]: No problem! It takes some effort to cover all the possibilities. Thanks for taking the time to look into this.

**Comment by tacaswell** [comment]
@MerlinSmiles You have to get the issue number right for the github magic to work ;)

**Comment by MerlinSmiles** [comment]
@tacaswell I might be very confused at the moment :)

**Pull request merged**